src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ ISC License
+
+ Copyright (c) 2016, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#include <string.h>
+#include "centerOfBrightness.h"
+
+/*! The constructor for the CenterOfBrightness module. It also sets some default values at its creation.  */
+CenterOfBrightness::CenterOfBrightness()
+{
+    this->filename = "";
+    this->saveImages = 1;
+    this->blurrSize = 5;
+    this->threshold = 50;
+    this->saveDir = "";
+}
+
+/*! This is the destructor */
+CenterOfBrightness::~CenterOfBrightness()
+{
+    return;
+}
+
+/*! This method performs a complete reset of the module.  Local module variables that retain time varying states between function calls are reset to their default values.
+ @return void
+ @param CurrentSimNanos The clock time at which the function was called (nanoseconds)
+ */
+void CenterOfBrightness::Reset(uint64_t CurrentSimNanos)
+{
+    // check that the required message has not been connected
+    if (!this->imageInMsg.isLinked()) {
+        bskLogger.bskLog(BSK_ERROR, "CenterOfBrightness.imageInMsg wasn't connected.");
+    }
+}
+
+/*! This module reads an OpNav image and extracts the weighted center of brightness. It performs a greyscale, a blurr,
+ * and a threshold on the image before summing the weighted pixel intensities in order to average them with the
+ * total detected intensity. This provides the center of brightness measurement (as well as the total number of
+ * bright pixels)
+ @return void
+ @param CurrentSimNanos The clock time at which the function was called (nanoseconds)
+ */
+void CenterOfBrightness::UpdateState(uint64_t CurrentSimNanos)
+{
+    std::string dirName;
+    CameraImageMsgPayload imageBuffer;
+    OpNavCOBMsgPayload cobBuffer;
+
+    imageBuffer = this->imageInMsg.zeroMsgPayload;
+    cobBuffer = this->opnavCOBOutMsg.zeroMsgPayload;
+
+    cv::Mat imageCV, blurred, imageGray;
```

Comment:
:large_orange_diamond: This variable should be declared just above its first use.

Reply:
Done